src/IR/HM.hs

```diff
@@ -107,12 +107,24 @@ runInferFn (InferFn m) = evalStateT
 inferProgram :: I.Program Ann.Type -> Compiler.Pass (I.Program Classes.Type)
 inferProgram p = runInferFn $ do
   typeDefs' <- recordADTs $ I.typeDefs p
+  recordFDefs $ I.programDefs p
```

Comment:
This is only called for the top-level, right? So this doesn't work for nested co-recursive functions?

Reply:
Do you mean nested co-recursive functions defined with let-bindings? No, these are not supported at this point.